core-data/index.js

```diff
@@ -0,0 +1,37 @@
+/**
+ * WordPress Dependencies
+ */
+import {
+	registerReducer,
+	registerSelectors,
+	registerResolvers,
+} from '@wordpress/data';
+
+/**
+ * Internal dependencies
+ */
+import reducer from './reducer';
+import { getCategories } from './selectors';
+
+/**
+ * Module Constants
+ */
+const MODULE_KEY = 'core';
```

Comment:
We need to establish some conventions around module names, both internally and as recommendations for plugins.

Elsewhere we prefix with a `core/` namespace and align the remainder of the name with the top-level folder. Here, that would mean `core/core-data`.

Obviously there's some ugliness in the repetition here. @mtias mentioned this at one point as well. My first thought was that we could drop the namespace (or make optional) for core modules, similar to what we do with blocks.

But then for plugins, we need to make certain that they _are_ namespaced. But for many plugins, they may only have a single store. What are they to call their stores? `yoast/yoast` ?

While verbose, at least the core namespace enables plugin authors to avoid their own, while avoiding potential name conflicts.

Whatever we decide, we should also be mindful of consistency with other "namespace + name" patterns.

Reply:
I think we could have a special case here because this `core-data` module will contain all "core" data, it could even be included in the data module itself if we didn't want to keep the data module generic. But happy to follow any consensus here.